### PR TITLE
Implement logging format configuration

### LINKED ISSUE #3177: Feature toggle to enable or disable JSON format logging
### Detailed description

In order to specify what kind of logging format we would like to see, we need a setting. During development, you'd like to see a human readable (and colored) output, whereas in production or other environments, you might want to format the output to JSON, to enable easier log aggregation.

This feature would be a continuation of #3114 

### Specifications

The implementation should include:

- [ ] An environment key `JSON_LOGGING` per module to enable JSON (dict) logging (see Mula)

### Additional information

The processors used by `structlog` should be reconsidered. For example, use color and key-value rendering for non JSON logging, and disable colors for the JSON output. This could otherwise result in color symbols in the JSON strings.

## Implementation

- Introduce `JSON_LOGGING` environment key (boolean, default `False`)
- Adapt processors in `structlog.configure` calls

### Alternatives considered

An alternative would be to always output JSON format logging and use a log aggregator and monitor (even during development). However, this approach adds complexity, since you'd be dependent on such tools, and it otherwise it might be harder to read the logs during development without these tools.

### PR DESCRIPTION
**Pull request by ammar92**
### Changes

This PR implements a logging format option. It is implemented as an environment variable with the possible options "text" and "json" for each service (e.g. `OCTOPOES_LOGGING_FORMAT`, with the exception of Rocky, which just uses `LOGGING_FORMAT`)

This also replaces the Mula configuration in the same style (`MULA_LOGGING_JSON` is now replaced with `MULA_LOGGING_FORMAT`).

Also note that environment settings documentation is automatically generated.

This PR depends on #3175 (is also branches of `feature/structured-logging`)

### Issue link

Closes #3177 

### Demo

image

image

### QA notes

Toggle the logging format option between *text* and *json* to see difference.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by underdarknl** [comment]
N.B. this needs a note in the release notes for 1.17, and Docs update

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Had some issues finding the right `.env` flags since I couldn't access the docs. However it works. We added the following flags to `.env` and you can see json formatted logs. 

```
OCTOPOES_LOGGING_FORMAT
BYTES_LOGGING_FORMAT
BOEFJES_LOGGING_FORMAT (for boefjes, normalizer en KATalogus)
SCHEDULER_LOGGING_FORMAT 
LOGGING_FORMAT (for Rocky)
```


### What doesn't work:
n/a

### Bug or feature?:
n/a

File: bytes/bytes/api/__init__.py [hunk 1]
```diff
@@ -39,16 +45,19 @@
 
 app = FastAPI(title="Bytes API")
 
-if get_settings().span_export_grpc_endpoint is not None:
-    logger.info("Setting up instrumentation with span exporter endpoint [%s]", get_settings().span_export_grpc_endpoint)
+if settings.span_export_grpc_endpoint is not None:
+    logger.info(
+        "Setting up instrumentation with span exporter endpoint [%s]",
+        get_settings().span_export_grpc_endpoint,
```
Comment (dekkers) [inline bytes/bytes/api/__init__.py:51]:
```suggestion
        settings.span_export_grpc_endpoint,
```

**Review by dekkers — APPROVED** [review]

**Comment by r3boot** [comment]
This PR does not implement JSON logging. After applying the needed changes (and focussing on two specific lines of logging that the boefje container outputs), we get the following logging output:

```
boefje-1  | [2024-08-20 13:00:12 +0000] [1] [INFO] [_client] HTTP Request: POST http://scheduler:8000/queues/boefje-z-cert/pop "HTTP/1.1 200 OK"
boefje-1  | [2024-08-20 13:00:12 +0000] [1] [INFO] [app] {"event": "Handling task[d1c196ef-7b91-458a-b93c-da1b105a292c]", "level": "info", "timestamp": "2024-08-20T13:00:12.848569"}
```

If we strip off the first column (container name), we get the output that would be put on stdout:

```
[2024-08-20 13:00:12 +0000] [1] [INFO] [_client] HTTP Request: POST http://scheduler:8000/queues/boefje-z-cert/pop "HTTP/1.1 200 OK"
[2024-08-20 13:00:12 +0000] [1] [INFO] [app] {"event": "Handling task[d1c196ef-7b91-458a-b93c-da1b105a292c]", "level": "info", "timestamp": "2024-08-20T13:00:12.848569"}
```

None of this output is JSON, but some of the lines do contain JSON. What we expect is the following (exact wording of the fields is up to KAT ofc):

```
{
    "timestamp": "2024-08-20 13:00:12 +0000",
    "number": 1,
    "level": "INFO",
    "name": "_client",
    "kind": "HTTP Request",
    "method": "POST",
    "url": "http://scheduler:8000/queues/boefje-z-cert/pop",
    "http_version": "HTTP/1.1",
    "status": 200,
    "message": "OK"
}
```

and

```
{
    "timestamp": "2024-08-20 13:00:12 +0000",
    "number": 1,
    "level": "INFO",
    "name": "app",
    "event": {
        "task": "d1c196ef-7b91-458a-b93c-da1b105a292c",
        "level": "info",
        "timestamp": "2024-08-20T13:00:12.848569"
    }
}
```

**Pull request merged**